service/matching/taskQueueManager.go

```diff
@@ -312,8 +312,10 @@ func (c *taskQueueManagerImpl) WaitUntilInitialized(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
-	_, verErr := c.metadataInitialFetchFut.Get(ctx)
-	return verErr
+	// We don't really care if the initial fetch worked or not, anything that *requires* a bit of metadata should fail
```

Comment:
just to get this straight, the goal is: block on the initial fetch request. if it succeeded, great. if it failed, let this call return and let the caller interact with the tqm anyway. the caller should do additional checks for metadata it needs? suppose the caller wants versioning data. how does it tell a successful fetch with no data from a failed fetch?

Reply:
The methods will all error if the data has never been successfully retrieved. If it has (and there is no data) then it won't error, and will return empty data.